internal/biz/resources/resources.go

```diff
@@ -82,14 +82,7 @@ func (uc *Usecase) Upsert(ctx context.Context, m *model.Resource, wait_for_sync
 	log.Info("upserting resource: ", m)
 	ret := m // Default to returning the input model in case persistence is disabled
 	var subscription pubsub.Subscription
-
-	// Generate txid for data layer
-	// TODO: Replace this when inventory api has proper api-level transaction ids
-	txid, err := uuid.NewV7()
-	if err != nil {
-		return nil, err
-	}
-	txidStr := txid.String()
+	var txidStr string
```

Comment:
Just to ensure understanding, before we always generated a txID and now we are only generating one if readAfterWrite is enabled? For instance on line 110, it would have a blank transaction id because we are updating an existing resource

Reply:
Okay yup, reread your summary. Makes sense!